src/implementations/title-escrow/reject-surrendered.ts

```diff
@@ -0,0 +1,67 @@
+import { TitleEscrowFactory, TradeTrustErc721Factory } from "@govtechsg/token-registry";
+import signale from "signale";
+import { getLogger } from "../../logger";
+import { getWallet } from "../utils/wallet";
+import { TitleEscrowSurrenderDocumentCommand } from "../../commands/title-escrow/title-escrow-command.type";
+import { dryRunMode } from "../utils/dryRun";
+import { TransactionReceipt } from "@ethersproject/providers";
+import { TradeTrustERC721 } from "@govtechsg/token-registry/dist/ts/contracts";
+import { Wallet } from "ethers";
+
+const { trace } = getLogger("token-registry:acceptSurrendered");
+
+export const retrieveLastBeneficiaryAndHolder = async (
+  tokenRegistryInstance: TradeTrustERC721,
+  tokenId: string,
+  wallet: Wallet
+): Promise<{ lastBeneficiary: string; lastHolder: string }> => {
+  // Fetch transfer logs from token registry
+  const transferLogFilter = tokenRegistryInstance.filters.Transfer(null, null, tokenId);
+  const logs = await tokenRegistryInstance.queryFilter(transferLogFilter, 0);
+  const lastTitleEscrowAddress = logs[logs.length - 1].args?.[0];
+  const lastTitleEscrowInstance = await TitleEscrowFactory.connect(lastTitleEscrowAddress, wallet);
+  const lastBeneficiary = await lastTitleEscrowInstance.beneficiary();
+  const lastHolder = await lastTitleEscrowInstance.holder();
```

Comment:
the getters are exposed like this in solidity?

i thought the getters were `getBeneficiary` and `getHolder`?

Reply:
til